src/pykeen/metrics/ranking.py

```diff
@@ -1123,8 +1123,10 @@ def harmonic_variances(n: int) -> np.ndarray:
     """
     h = generalized_harmonic_numbers(n)
     h2 = generalized_harmonic_numbers(n, p=-2)
-    n = np.arange(1, n + 1)
+    n = np.arange(1, n + 1, dtype=float)
     v = (n * h2 - h**2) / n**2
+    # ensure non-negativity
+    v = np.maximum(v, 0.0)
```

Comment:
how could this number ever be negative? just because of a numerical artifact?

Reply:
Mathematically, I cannot be. But since we are subtracting two floating point numbers in the line above, it could happen, cf. e.g. [here](https://en.wikipedia.org/wiki/Loss_of_significance#Subtraction).

Reply:
I put this explanation into the code, too, cf. 5d238c7